### PR TITLE
Migrate fast double parser to conan 2.x

### PR DESCRIPTION
**Pull request by AlexandrePTJ**
Following the migration webinar, here is the migration for fast_double_parser library

### CODE REVIEW

File: recipes/fast_double_parser/all/conanfile.py [hunk 1]
```diff
@@ -1,5 +1,8 @@
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import copy, get
+from conan.tools.layout import basic_layout
 import os
-from conans import ConanFile, tools
 
 required_conan_version = ">=1.43.0"
```
Comment (uilianries) [inline recipes/fast_double_parser/all/conanfile.py:7]:
```suggestion
required_conan_version = ">=1.50.0"
```
check_min_cppstd: https://github.com/conan-io/conan-center-index/blob/master/docs/v2_linter.md

File: recipes/fast_double_parser/all/conanfile.py [hunk 2]
```diff
@@ -15,22 +18,22 @@ class FastDoubleParserConan(ConanFile):
...
     no_copy_source = True
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def layout(self):
+        basic_layout(self)
 
     def validate(self):
+        print(self.info.__dict__)
```
Comment (uilianries) [inline recipes/fast_double_parser/all/conanfile.py:25]:
```suggestion
```

File: recipes/fast_double_parser/all/test_v1_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+find_package(fast_double_parser REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} fast_double_parser::fast_double_parser)
+target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_11)
```
Comment (uilianries) [inline recipes/fast_double_parser/all/test_v1_package/CMakeLists.txt:11]:
```suggestion
add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../test_package/
                 ${CMAKE_CURRENT_BINARY_DIR}/test_package/)
```
Re-use CMakeLists.txt from test_package

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution! For future migration PRs, you can consider https://github.com/conan-io/conan-center-index/tree/master/docs/package_templates as good templates.

**Review by MartinDelille — APPROVED** [review]

**Comment by uilianries** [comment]
@AlexandrePTJ Could please ask to your include your user in the authorized list: https://github.com/conan-io/conan-center-index/pull/13982#issuecomment-1305210891

**Comment by MartinDelille** [comment]
@AlexandrePTJ You might need to close and reopen this PR to trigger the CI again.

**Comment by MartinDelille** [comment]
You just need to reopen this one. No need to open another one (https://github.com/conan-io/conan-center-index/pull/14199)

**Comment by AlexandrePTJ** [comment]
@MartinDelille Ok, sorry. So I close #14199 and re-open this one.

**Review by prince-chrismc — APPROVED** [review]

**Review by MartinDelille — APPROVED** [review]

**Pull request merged**